api/api.go

```diff
@@ -473,7 +523,15 @@ func NewClient(config *Config) (*Client, error) {
 
 	httpClient := config.HttpClient
 	if httpClient == nil {
-		httpClient = defaultHttpClient()
+		addrURL, _ := url.Parse(config.Address)
```

Comment:
Not a show-stopper, as some of the behavior is existing...

If we're not using the default config and `config.Address` doesn't parse, we'll have caught the error above but also we'll have parsed the address twice. And then we always parse it again on `newRequest` (that's existing behavior). Other than setter methods for region and namespace, the consuming code can't change the config of a client (because `Client.config` isn't exported). Maybe we could parse the address once here and then stash the resulting [`net/url.URL`](https://pkg.go.dev/net/url#URL) struct in a private field of either `Client` or `Config`?

Reply:
I took a swing at reducing the calls to `url.Parse` in [910e06e](https://github.com/hashicorp/nomad/pull/16872/commits/910e06e479b32440a6fb6600d9a910886c413d00). Would appreciate your thoughts on it.